DetectRange.c

```diff
@@ -0,0 +1,80 @@
+#include "BMS_CurRead.h"
+
+
+#include <stdio.h>
+
+int NumberofContRange=0;
+
+RangeReading ArrayofContRange[]={};
+int NumberofContinuousRange(int *ChargingCurrentReadings, int numOfCCReadings)
+{
+	int l_NumberofContRange=NumberofContRange;
+	NumberofContRange=0;
+	return l_NumberofContRange;
+}
+	
+void swap(int *xp, int *yp)
+{
+    float temp = *xp;
+    *xp = *yp;
+    *yp = temp;
+}
+
+RangeReading printFunction( int a_StartIndex, int a_EndIndex, int a_NumberinRange)
+{
+	RangeReading l_RangeReading;
+	l_RangeReading.StartIndex=a_StartIndex;
+	l_RangeReading.EndIndex= a_EndIndex;
+	l_RangeReading.NumOfReadingsInRange= a_NumberinRange;
+	 printf("The number of readings in the range %d - %d is %d",  a_StartIndex, a_EndIndex , a_NumberinRange);
+	 
+	 return l_RangeReading;
+}
+
+void SortArray(int *ChargingCurrentReadings, int numOfCCReadings)

```

Comment:
why not use stdlib qsort function? Otherwise you need to test the sort functionality!